src/main/kotlin/com/terraformation/backend/funder/db/PublishedProjectDetailsStore.kt

```diff
@@ -0,0 +1,177 @@
+package com.terraformation.backend.funder.db
+
+import com.terraformation.backend.accelerator.model.CarbonCertification
+import com.terraformation.backend.accelerator.model.SustainableDevelopmentGoal
+import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.db.asNonNullable
+import com.terraformation.backend.db.default_schema.LandUseModelType
+import com.terraformation.backend.db.default_schema.ProjectId
+import com.terraformation.backend.db.funder.tables.records.PublishedProjectDetailsRecord
+import com.terraformation.backend.db.funder.tables.references.PUBLISHED_PROJECT_CARBON_CERTS
+import com.terraformation.backend.db.funder.tables.references.PUBLISHED_PROJECT_DETAILS
+import com.terraformation.backend.db.funder.tables.references.PUBLISHED_PROJECT_LAND_USE
+import com.terraformation.backend.db.funder.tables.references.PUBLISHED_PROJECT_SDG
+import com.terraformation.backend.funder.event.FunderProjectProfilePublishedEvent
+import com.terraformation.backend.funder.model.FunderProjectDetailsModel
+import jakarta.inject.Named
+import java.math.BigDecimal
+import java.time.InstantSource
+import org.jooq.DSLContext
+import org.springframework.context.ApplicationEventPublisher
+
+@Named
+class PublishedProjectDetailsStore(
+    private val clock: InstantSource,
+    private val dslContext: DSLContext,
+    private val eventPublisher: ApplicationEventPublisher,
+) {
+  fun fetchOneById(projectId: ProjectId): FunderProjectDetailsModel? {
+    // convert the list to a set
+    val sdgList =
+        with(PUBLISHED_PROJECT_SDG) {
+          dslContext
+              .select(SDG_NUMBER.asNonNullable())
+              .from(this)
+              .where(PROJECT_ID.eq(projectId))
+              .fetchSet(SDG_NUMBER.asNonNullable())
+              .mapNotNull { SustainableDevelopmentGoal.bySdgNumber[it] }
+              .toSet()
+        }
+    val carbonCerts =
+        with(PUBLISHED_PROJECT_CARBON_CERTS) {
+          dslContext
+              .select(CARBON_CERTIFICATION.asNonNullable())
+              .from(this)
+              .where(PROJECT_ID.eq(projectId))
+              .fetchSet(CARBON_CERTIFICATION.asNonNullable())
+              .mapNotNull { CarbonCertification.forDisplayName(it) }
```

Comment:
I realize this is part of the data model that came before this PR, but I'm a little unclear on the semantics here. We're storing the carbon certification as text, but then requiring it to be a valid enum value? Why aren't we storing an integer enum ID with a foreign key reference to an enum table to guarantee only valid values? What's the advantage of storing a display name here?

Reply:
No advantage, but I'm not familiar with anywhere in the app that references the select options of a docprod variable. Carbon Certifications is a multi select variable (as are land use model type and sdg), so if we do what you're saying then we would need to reference only some rows of `docprod.variable_select_options`. Have we done something like that before?